internal/experiments/experiments_remote_execute_api.go

```diff
@@ -0,0 +1,205 @@
+package experiments
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/operantai/secops-chaos/internal/categories"
+	"github.com/operantai/secops-chaos/internal/executor"
+	"github.com/operantai/secops-chaos/internal/k8s"
+	"github.com/operantai/secops-chaos/internal/verifier"
+	"gopkg.in/yaml.v3"
+)
+
+type RemoteExecuteAPIExperimentConfig struct {
+	Metadata   ExperimentMetadata `yaml:"metadata"`
+	Parameters RemoteExecuteAPI   `yaml:"parameters"`
+}
+
+// RemoteExecuteAPI is an experiment that uses the remote executor to check a remote output
+// The image must be created independently -- the current default is `alconen/egress_server`, which runs a simple web app on port 4000 that checks http connectivity to
+// a few domains ("https://google.com", "https://linkedin.com", "https://openai.com/") and responds with a success based on the success of those calls.
+// The source can be found at internal/executor/container_examples/egress_server
+type RemoteExecuteAPI struct {
+	Image                string `yaml:"image"`
+	ImageParameters      string `yaml:"image_parameters"`
+	TargetPort           int32  `yaml:"target_port"`
+	Path                 string `yaml:"path"`
+	ServiceAccountName   string `yaml:"service_account_name"`
+	LocalPort            int32  `yaml:"local_port"`
+	ExpectedHTTPResponse int    `yaml:"expected_response"`
+	ExpectedBody         string `yaml:"expected_body"`
+}
```

Comment:
This is probably more personal preference but I'd do something like:
```suggestion
type RemoteExecuteAPI struct {
	Image                string `yaml:"image"`
	ImageParameters      string `yaml:"image_parameters"`
	ServiceAccountName string `yaml:"service_account_name"`
	Target Target `yaml:"target"`
}

type Target struct {
	Port           int32  `yaml:"target_port"`
	Path                 string `yaml:"path"`
	ExpectedHTTPResponse int    `yaml:"expected_response"`
	ExpectedBody         string `yaml:"expected_body"`
}
```

Remove LocalPort, and use `:0` instead, it will select a random open port. That way people don't have to mess around trying to find an open port locally. You can read more about it [here](https://stackoverflow.com/questions/43424787/how-to-use-next-available-port-in-http-listenandserve)